api/app/src/test/kotlin/packit/integration/controllers/RoleControllerTest.kt

```diff
@@ -143,4 +144,92 @@ class RoleControllerTest : IntegrationTest()
         val role = roleRepository.findByName("testRole")!!
         assertEquals(0, role.rolePermissions.size)
     }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["user.manage"])
+    fun `users with manage authority can get role names`()
+    {
+        roleRepository.save(Role(name = "testRole"))
+
+        val result = restTemplate.exchange(
+            "/role",
+            HttpMethod.GET,
+            getTokenizedHttpEntity(),
+            String::class.java
+        )
+
+        assertSuccess(result)
+
+        assertEquals(ObjectMapper().writeValueAsString(listOf("ADMIN", "testRole")), result.body)
+    }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["user.manage"])
+    fun `users can get roles with relationships`()
+    {
+        val roleDto = roleRepository.findByName("ADMIN")!!.toDto()
+        val result = restTemplate.exchange(
+            "/role/complete",
+            HttpMethod.GET,
+            getTokenizedHttpEntity(),
+            String::class.java
+        )
+
+        assertSuccess(result)
+
+        assertEquals(ObjectMapper().writeValueAsString(listOf(roleDto)), result.body)
+    }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["user.manage"])
+    fun `users can get username roles with relationships `()
+    {
+        roleRepository.save(Role("username", isUsername = true))
+        val allUsernameRoles = roleRepository.findAllByIsUsername(true).map { it.toDto() }
+        val result = restTemplate.exchange(
+            "/role/complete/usernames",
+            HttpMethod.GET,
+            getTokenizedHttpEntity(),
+            String::class.java
+        )
+
+        assertSuccess(result)
+
+        assertEquals(ObjectMapper().writeValueAsString(allUsernameRoles), result.body)
+    }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["user.manage"])
+    fun `users can get non username roles with relationships`()
+    {
+        roleRepository.save(Role("randomUser", isUsername = true)).toDto()
+        val adminRole = roleRepository.findByName("ADMIN")!!.toDto()
```

Comment:
Isn't there a bit of a dangerous assumption in this sort of test that your repository is functioning correctly? It might not be pulling all values out correctly, in which case we'd just be comparing the wrong values from direct repo call with the wrong values from the endpoint and passing the test. We're not testing the repository elsewhere than in these integration tests, so that could happen. It feels like the test objects to compare against should be defined in the test itself. You'll need to pull the ids out of the db, but the rest can be hardcoded.

Reply:
as discussed have created ticket for repository tests